demos/vc_issuer/src/main.rs

```diff
@@ -150,20 +150,12 @@ fn authorize_vc_request(
     alias: &SignedIdAlias,
     current_time_ns: u128,
 ) -> Result<AliasTuple, IssueCredentialError> {
-    let alias_tuple = extract_id_alias(alias, current_time_ns)?;
-
-    if caller() != alias_tuple.id_dapp {
-        return Err(IssueCredentialError::UnauthorizedSubject(format!(
-            "Caller {} does not match id alias dapp principal {}.",
-            caller(),
-            alias_tuple.id_dapp
-        )));
-    }
-    Ok(alias_tuple)
+    extract_id_alias(alias, &caller(), current_time_ns)
```

Comment:
Should we also remove this indirection and maybe inline `extract_id_alias`?
It would also be nice to handle `caller()` and `time()` the same way (i.e. either pass both as an argument or call the system API in this function).